components/mpas-ocean/src/analysis_members/mpas_ocn_harmonic_analysis.F

```diff
@@ -743,18 +743,21 @@ subroutine least_squares_decompose(nfreq,ha) !{{{
 
 !     Decomposition of matrix 
 
-      do 100 ir=1,mm
+      do ir=1,mm
          ire=ir+1
-         do 20 j=ire,mm
- 20         ha(ir,j)=ha(ir,j)/ha(ir,ir)
-         if (ire.gt.mm) goto 100
-         do 40 j=ire,mm
-           do 40 k=ire,mm
- 40           ha(k,j)=ha(k,j)-ha(k,ir)*ha(ir,j)
-         do 50 j=ire,mm
- 50        ha(j,ir)=0.0_RKIND
- 100  continue
-
+         do j=ire,mm
+            ha(ir,j)=ha(ir,j)/ha(ir,ir)
+         end do
+         if (ire.gt.mm) cycle
```

Comment:
@mark-petersen, I think `cycle` skips the remainder of the loop iteration. I believe we want an `exit` here to get the same behavior as the `goto 100` statement.

Reply:
For the general case of a `goto` to the `continue` line of a loop, I believe `cycle` is correct. But for this specific logic, exit would do the same thing, because the test `(ire.gt.mm)` is only true on the last iteration anyway. Yes, I can change it. I think that is more clear.

Reply:
corrected